### PR TITLE
Replace product task with custom completion logic

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Changes proposed in this Pull Request:

Closes https://github.com/Automattic/wc-calypso-bridge/issues/955.

This PR adds a replacement product class that uses a custom completion logic. Applies to all plans.

However, this PR does not address if the user modifies the sample product, it'll simply not be counted as a published product.

### How to test the changes in this Pull Request:

1. Make sure there are no products in your site
2. Import products [CSV from headstart](https://gist.githubusercontent.com/ilyasfoo/1907a9d5e5b8a72fefe2af6e8d05ef03/raw/c3f3e5e2a52169480b9aeab6e1ec8e672fbb73db/headstart-product-export.csv) 
3. Go to WooCommerce > Home, observe that product task is not marked as completed
5. Add a new product and publish it
6. Go to WooCommerce > Home, observe that product task is marked as completed

### CODE REVIEW

File: includes/tasks/class-wc-calypso-task-headstart-products.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		return self::has_products();
+	}
+
+	/**
+	 * Check if the store has any published products,
+	 * but exclude sample products generated by Headstart themes.
+	 *
+	 * @return bool
+	 */
+	public static function has_products() {
```
Comment (chihsuan) [inline includes/tasks/class-wc-calypso-task-headstart-products.php:27]: @ilyasfoo I think we could just use one query to get the same outcome. What do you think?

```php
public static function has_products() {
	$product_query = array(
		'post_type' => 'product',
		'post_status' => 'publish',
		'fields' => 'ids',
		'meta_query' => array(
			array(
				'key' => '_headstart_post',
				'compare' => 'NOT EXISTS'
			)
		)
	);
	$products = get_posts($product_query);
	$product_count = count($products);

	return $product_count > 0;
}
```

**Review by chihsuan — APPROVED** [review]
Looks good and tested well. 💯

**Review by daledupreez — COMMENTED** [review]
Without testing, this looks perfectly reasonable to me. Let me know if you'd like me to test as well, @ilyasfoo.

In terms of next steps, might it make sense to create a follow-up task to cleanly handle the case where users modify these products? I am happy to create the issue -- I am thinking that the implementation could also pull the `post_date_gmt` and `post_modified_gmt` fields in the query and remove any products with a post->modified gap larger than 60 seconds (or some similarly generous window).

But I strongly think that belongs in a follow-up PR so we can get this one shipped.

File: includes/tasks/class-wc-calypso-task-headstart-products.php (hunk 1 above)
Comment (ilyasfoo) [reply includes/tasks/class-wc-calypso-task-headstart-products.php:27]: Actually, you have a point here. `get_posts` can be set to return only the post ids, which means the data return size would be negligible.

In contrast, your solution would only require a single round trip to db which is better.  I'm leaning towards making the change

**Comment by ilyasfoo** [comment]
> Without testing, this looks perfectly reasonable to me. Let me know if you'd like me to test as well, @ilyasfoo.

@daledupreez Thanks, no need for another test 👍 

> In terms of next steps, might it make sense to create a follow-up task to cleanly handle the case where users modify these products? I am happy to create the issue

Yes that sounds good, thanks!

**Comment by daledupreez** [comment]
I've created a follow-up issue in #973.

**Review by chihsuan — APPROVED** [review]
Thanks for making the change. LGTM! 💜

**Pull request merged**